vega_sim/scenario/common/agents.py

```diff
@@ -1828,21 +2168,63 @@ def step(self, vega_state: VegaState):
                 [order.volume for order in order_book.buys if order.price > next_price]
             )
 
-        volume = round(self.proportion_taken * volume, self.pdp)
+        # Determine the size of the order
+        size = round(self.proportion_taken * volume, self.pdp)
 
-        if (not market_in_auction) and volume:
-            try:
-                self.vega.submit_market_order(
-                    trading_wallet=self.wallet_name,
-                    market_id=self.market_id,
-                    side=trade_side,
-                    volume=volume,
-                    wait=False,
-                    fill_or_kill=False,
-                    key_name=self.key_name,
-                )
-            except OrderRejectedError:
-                logger.debug("Order rejected")
+        # Add a random probability the agent speculates the wrong side
+        if self.random_state.rand() <= self.accuracy:
+            side = side
+        else:
+            side = (
+                vega_protos.SIDE_BUY
+                if side == vega_protos.SIDE_SELL
+                else vega_protos.SIDE_SELL
+            )
+
+        # Attempt to submit the order and add it to the queue
+        try:
+            self.vega.submit_market_order(
+                trading_wallet=self.wallet_name,
+                market_id=self.market_id,
+                side=side,
+                volume=size,
+                wait=False,
+                fill_or_kill=False,
+                key_name=self.key_name,
+            )
+            return ITOrder(side=side, size=size)
+
+        except OrderRejectedError:
+            logger.debug("Order rejected")
+            return None
+
+    def _settle_order(self, order: ITOrder):
```

Comment:
Interesting setup. So 'settle' here means the agent closing out the amount of position it took from the matching original position? Seems like a good plan. A couple of changes I'd probably make would be making the queue length configurable (or maybe even make a max 'allowed abs position' before running the closeouts too?) and rename to something slightly clearer about what's going on, I'm thinking `_close_positions` or something?

Reply:
> So 'settle' here means the agent closing out the amount of position it took from the matching original position?
Yeah exactly, very happy to change the name of the method, `_close_positions` sounds good.

So the queue length is configured through the `lookahead` arg, as we only get an order from the queue when the queue is full we know when we do the correct number of steps has passed (why its also important to add a "fake" order to the queue even if no order is placed).

This does break however if a market goes into auction because obvs market orders no longer work... open to ideas on this, was maybe thinking once it exits auctions it closes its position and empties the cue.

Reply:
Makes sense on `lookahead`. And yeah seems reasonable to just wait if the market is in auction and close out when it reopens since this agent isn't really trying to do anything special whilst in auction.